cpp/src/io/json/write_json.cu

```diff
@@ -86,6 +86,37 @@ struct escape_strings_fn {
       bytes += cudf::strings::detail::bytes_in_char_utf8(chr);
   }
 
+  __device__ inline char nibble_to_hex(uint8_t nibble) const
```

Comment:
We reinvent functions similar to this one several times. Is reuse possible?

https://github.com/rapidsai/cudf/blob/deec3f8f981fd89f1aa46c6aea3714fd7c7355b9/cpp/src/strings/convert/convert_urls.cu#L60

https://github.com/rapidsai/cudf/blob/deec3f8f981fd89f1aa46c6aea3714fd7c7355b9/cpp/src/strings/convert/convert_hex.cu#L135

https://github.com/rapidsai/cudf/blob/deec3f8f981fd89f1aa46c6aea3714fd7c7355b9/cpp/include/cudf/detail/utilities/hash_functions.cuh#L170

Reply:
`uint32ToLowercaseHexString` is interesting.  It's similar algorithm to instrinsics suggestion but without comparison (comparison achieved adding 6 & ANDing higher order bit). But uint32 is not required here. Only uint16 is required here.
Reusing `byte_to_hex` is possible. IMO, while working directly with nibbles and bytes, I prefer hex usage instead of numbers `16`, division and subtractions.